python/nav/eventengine/severity.py

```diff
@@ -0,0 +1,221 @@
+#
+# Copyright (C) 2021 Uninett AS
+#
+# This file is part of Network Administration Visualized (NAV).
+#
+# NAV is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License version 3 as published by
+# the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.  You should have received a copy of the GNU General Public
+# License along with NAV. If not, see <http://www.gnu.org/licenses/>.
+#
+"""Contains functionality for eventengine to parse user-configured alert severity
+modification rules and apply them to generated alerts.
+
+To load a set of rules from a YAML config file, you can do:
+
+    rules = SeverityRules.load_from_file("my_config_file.yml")
+
+Then, given an alert object you wish to assign a new severity value to:
+
+    new_severity = rules.evaluate(alert)
+    alert.severity = new_severity
+    alert.save()
+
+"""
+from operator import attrgetter
+import logging
+import typing
+import yaml
+
+from nav import config
+from nav.models import event
+
+__all__ = ["Severity", "SeverityRules"]
+#
+# Module constants
+#
+
+
+DEFAULT_SEVERITY = 3  # middle of the road
```

Comment:
If what this code does is globally defining the default severity level as 3, this should be documented.
If not, please ignore this comment.

Reply:
In fact, this constant is only used if someone tries to evaluate an object that doesn't have a severity attribute attached at all (i.e. you are evaluating an object that *is not an alert object*, either in error, or because you are mocking alert objects in your test code).